packages/aws-cdk-lib/aws-ecs-patterns/lib/fargate/application-load-balanced-fargate-service.ts

```diff
@@ -152,4 +174,28 @@ export class ApplicationLoadBalancedFargateService extends ApplicationLoadBalanc
       throw new Error(`${name}: Must be a non-negative integer; received ${value}`);
     }
   }
+
+  private validateContainerCpu(containerCpu?: number, cpu: number = 256) { // default value for cpu is 256
+    if (containerCpu === undefined || Token.isUnresolved(containerCpu) || Token.isUnresolved(cpu)) {
+      return;
+    }
+    if (containerCpu > cpu) {
+      throw new Error(`containerCpu must be less than to cpu; received containerCpu: ${containerCpu}, cpu: ${cpu}`);
+    }
+    if (containerCpu < 0 || !Number.isInteger(containerCpu)) {
+      throw new Error(`containerCpu must be a non-negative integer; received ${containerCpu}`);
+    }
+  }
+
+  private validateContainerMemoryLimitMiB(containerMemoryLimitMiB?: number, memoryLimitMiB: number = 512) { // default value for memoryLimitMiB is 512
+    if (containerMemoryLimitMiB === undefined || Token.isUnresolved(containerMemoryLimitMiB) || Token.isUnresolved(memoryLimitMiB)) {
+      return;
+    }
+    if (containerMemoryLimitMiB > memoryLimitMiB) {
+      throw new Error(`containerMemoryLimitMiB must be less than to memoryLimitMiB; received containerMemoryLimitMiB: ${containerMemoryLimitMiB}, memoryLimitMiB: ${memoryLimitMiB}`);
+    }
+    if (containerMemoryLimitMiB <= 0 || !Number.isInteger(containerMemoryLimitMiB)) {
+      throw new Error(`containerMemoryLimitMiB must be a positive integer; received ${containerMemoryLimitMiB}`);
+    }
```

Comment:
containerMemoryLimitMiB must be an integer greater than 0, so the error message specifies "Must be a positive integer"